### PR TITLE
[WIP] add github-review

### PR DESCRIPTION
**Pull request by Emiller88**
Pretty simple and cool package.
@hlissner I'm going to request you do the keybinding for
`github-review-forge-pr-at-point`, or shoot me an example of binding something
in a magit buffer, and what you'd like the binding to be.

### CODE REVIEW

**Comment by darth10** [comment]
IMHO a binding for `github-review-forge-pr-at-point` would fit quite well under the `magit-merge` menu as it's essentially means "review a merge".
I use this in my config:

```lisp
(transient-append-suffix 'magit-merge "i"
  '("y" "Review pull request" github-review-forge-pr-at-point))
```

**Comment by Emiller88** [comment]
@darth10 That looks good. My vote however would go under the `@` for forge, or I don't see what the harm in having both is.

File: modules/tools/magit/config.el [hunk 1]
```diff
@@ -115,6 +115,7 @@ ensure it is built when we actually use Forge."
           (dolist (hook forge-bug-reference-hooks)
             (add-hook hook #'forge-bug-reference-setup)))))))
 
+(use-package! github-review)
```
Comment (hlissner) [inline modules/tools/magit/config.el]: This unnecessary eager-loads the package. It should be both lazy loaded and where you place your transient bindings.

Comment (hlissner) [reply modules/tools/magit/config.el]: There should also be two blank lines on either side of this `use-package!` block.

**Comment by Emiller88** [comment]
@hlissner Your right it doesn't and that `+magit/start-github-review ` makes it work well outside a forge workflow now.

**Comment by hlissner** [comment]
Could you rebase this PR onto `develop`? Besides the conflict I think this looks ready to merge.

**Comment by Emiller88** [comment]
@hlissner should be good to go!

**Comment by hlissner** [comment]
Looks good to me! Thanks again for your help!

**Pull request merged**